rp/src/cli.rs

```diff
@@ -0,0 +1,458 @@
+use std::path::PathBuf;
+use std::{iter::Peekable, net::SocketAddr};
+
+use crate::exchange::{ExchangeOptions, ExchangePeer};
+
+pub enum Command {
```

Comment:
I am not sure if I understand the CLI parsing here, but I think that using clap would be a better idea, because we already use it for the core application and a consistent user experience would be nice :)

Reply:
Well, the goal was reimplementing the rp command and this is what this does.

Migrating to clap requires a CLI redesign and I think we should consult somewone to create a ux concept for that.

Lets not grow the scope of this

Reply:
Yeah, I mainly opted to custom-parsing everything because `clap` cannot handle the CLI argument format that the original `rp` used, which would break existing scripts/documentation.